Add preset for `climbing=route_bottom`

zbycz opened the pull request:
### Description, Motivation & Context

Hi, here is my another take on adding climbing features to iD. This time I focused only on what is already prevalent in OSM.  Last year I proposed a PR (#1113) which stalled - perhaps it was too large, or perhaps I add too much of unused stuff from OSM wiki.

In this first PR, I would like to add only `climbing=route_bottom` node preset + its fields.

### Preset: Climbing Route Start (`climbing=route_bottom`)

 - This tag is used to denote the position of the start of the climbing route. It is usually the most important to have in GPS.
 - This node usually holds all the information about the upcoming route. See [picture](https://github.com/user-attachments/assets/7dd89c39-dbf7-48ad-b92f-a87f2f256ff2) for clarification. 
 - There is a possible evolution, where node can be replace by `way` with `climbing=route`, but it is not as usual in OSM DB yet.

#### Linked fields from the preset

I wrote a script which measured all the tag combinations - see results [here](https://gist.github.com/zbycz/05895772588052c8743ce6f32da7c1c1). I propose to add only fields which are used more often and are already established on the OSM wiki.

| count | tag | label | description
|--|--|--|--|
| 15582 | `name=*` |  |
| 12631 | `climbing:grade:uiaa=*` | Grade (UIAA) | Different classification scales - typically used in different countries
| 911 | `climbing:grade:french=*` | Grade (French)
| 757 |`climbing:grade:saxon=*` | Grade (Saxon)
| 145 |`climbing:grade:yds_class=*` | Grade (YDS) | [Yosemite Decimal System](https://en.wikipedia.org/wiki/Yosemite_Decimal_System) - not that prevalent, but important
| 8866 | `website=*` | |
| 2323 | `climbing:length=#` | Length (m) | Length of the upcoming route - useful to have long enough rope
| 1661 | `climbing:sport=yes` | Sport climbing | Type of climbing, where the route is secured by pre-drilled bolts ([wikipedia](https://en.wikipedia.org/wiki/Sport_climbing)) 
| 1007 | `climbing:trad=yes` | Traditional climbing | Type of climbing, where climber has to place their own protection ([wikipedia](https://en.wikipedia.org/wiki/Traditional_climbing))
| 723 | `climbing:bolts=#` | Number of Bolts | Number of the pre-drilled bolts in the upcoming route.
| 284 | `description=*` | |

### Related issues

- https://github.com/openstreetmap/id-tagging-schema/issues/31

### Links and data

**Relevant OSM Wiki links:**
- https://wiki.openstreetmap.org/wiki/Climbing

**Relevant tag usage stats:**
- 15 582 nodes - https://taginfo.openstreetmap.org/tags/climbing=route_bottom


### Checklist and Test-Documentation Template

<details><summary>Read on to get your PR merged faster…</summary>

Follow these steps to test your PR yourself and make it a lot easier and faster for maintainers to check and approve it.

**This is how it works:**
1. After you submit your PR, the system will create a preview and comment on your PR:
   > 🍱 You can preview the tagging presets of this pull request here.
   If this is your first contribution to this project, the preview will not happen right away but requires a click from one of the project members. We will do this ASAP.

2. Once the preview is ready, use it to test your changes.

3. Now copy the snippet below into a new comment and fill out the blanks.

4. Now your PR is ready to be reviewed.

```
## Test-Documentation

### Preview links & Sidebar Screenshots

<!-- Use the preview to find examples, select the feature in question and **copy this link here**.
     Find examples of nodes/areas. Find examples with a lot of tags or very few tags. – Whatever helps to test this thoroughly.
     Add relevant **screenshots** of the sidebar of those examples. -->

<!-- FYI: What we will check:
     - Is the [icon](https://github.com/ideditor/schema-builder/blob/main/ICONS.md) well chosen.
     - Are the fields well-structured and have good labels.
     - Do the dropdowns (etc.) work well and show helpful data. -->

### Search

<!-- **Test the search** of your preset and share relevant **screenshots** here.
     - Test the preset name as search terms.
     - Also test the preset terms and aliases as search terms (if present). -->

### Info-`i`

<!-- **Test the info-i** for your fields and preset and share relevant **screenshots** here.
     The info needs to help mappers understand the preset and when to use it.
     [Learn more…](https://github.com/openstreetmap/id-tagging-schema/blob/main/CONTRIBUTING.md#info-i)
 -->

### Wording

- [ ] American English
- [ ] `name`, `aliases` (if present) use Title Case
- [ ] `terms` (if present) use lower case, sorted A-Z
<!-- Learn more in https://github.com/openstreetmap/id-tagging-schema/blob/main/GUIDELINES.md#2-design-the-preset -->
```

</details>

tordans commented:
> I added the wikibase entry, but it doesn't show. Instead it links to "key:climbing" (https://wiki.openstreetmap.org/wiki/Item:Q3036) How to link this new one https://wiki.openstreetmap.org/wiki/Item:Q23038 ?

You have to add a description in the data item table, eg  https://wiki.openstreetmap.org/wiki/Item:Q8221

> <img width="362" alt="image" src="https://github.com/user-attachments/assets/bfa38c7e-1cb5-45fa-bb4f-8464ed10c8e7" />

zbycz commented:
@tordans - thanks for a comprehensive review. I agree that this is a good oportunity to update the wiki. I will work on it this week. Once it is visible on wiki, i will respond to the comments with links.

ad link to Q230238) i dont understand your comment - this item has the label specified, but iD still ignores the dataitem altogether. It seems to only fetch a wiki page, not the item. Maybe it gets fixed after i create the route_bottom page? or do i have to link it to the item?

![image](https://github.com/user-attachments/assets/598b0b5a-6ef0-464b-ab5b-3906ebaa9a10)

musoke commented on data/fields/climbing/grade/french.json:
```diff
@@ -0,0 +1,6 @@
+{
+    "key": "climbing:grade:french",
+    "type": "text",
```
Grades are very diverse.   French grades might benefit from a combo, but other grades get more complicated.

zbycz commented on data/fields/climbing/length.json:
```diff
@@ -0,0 +1,6 @@
+    "key": "climbing:length",
+    "type": "number",
+    "minValue": 0,
+    "label": "Length (m)"
```
Thanks, fixed.

zbycz commented on data/fields/climbing/bolts.json:
```diff
@@ -0,0 +1,6 @@
+    "key": "climbing:bolts",
+    "type": "number",
+    "minValue": 0,
+    "label": "Number of Bolts"
```
That sounds good. Added and pushed the changes.

zbycz commented on data/fields/climbing/sport.json:
```diff
@@ -0,0 +1,5 @@
+{
+    "key": "climbing:sport",
+    "type": "check",
```
I created the pages with explanation. In short - the numbers are on different features (crags,areas):
- https://wiki.openstreetmap.org/wiki/Key:climbing:sport
- https://wiki.openstreetmap.org/wiki/Key:climbing:trad

zbycz commented on data/fields/climbing/sport.json:
```diff
@@ -0,0 +1,5 @@
+{
+    "key": "climbing:sport",
+    "type": "check",
+    "label": "Sport climbing"
```
I explored this topic little bit more, and usually the hypernym Rock climbing, consists of Sport climbing (with bolts+rope), Bouldering (shorter, without bolts), etc. The [wikipedia states it clearly](https://en.wikipedia.org/wiki/Sport_climbing#:~:text=While%20competition%20climbing%20consists%20of%20three%20distinct%20rock%20climbing%20disciplines%20of%20lead%20climbing%20(the%20bolted%20sport%20climbing%20element)%2C%20bouldering%20(no%20bolts%20needed)%2C%20and%20speed%20climbing%20(also%20not%20bolted%2C%20but%20instead%20top%20roped)%2C%20it%20is%20sometimes%20confusingly%20referred%20to%20as%20%22sport%20climbing%22.).

It seems the Olympics really added some confusion here.

To make it clearer, I added a bracket: `Sport climbing (bolted climbing)`  – specifically because we don't have the bouldering option (yet), so begginers could try to fit it wrongly. What do you think?

zbycz commented on data/presets/climbing/route_bottom.json:
```diff
@@ -0,0 +1,33 @@
+        "point",
+        "vertex"
+    ],
+    "fields": [
+        "name"
+    ],
+    "moreFields": [
+        "climbing/grade/uiaa",
+        "climbing/grade/french",
+        "climbing/grade/saxon",
+        "climbing/grade/yds_class",
+        "website",
+        "climbing/length",
+        "climbing/sport",
+        "climbing/trad",
+        "climbing/bolts",
+        "description"
+    ],
```
Yes, usually routes have names - I have never heard about a route without name. But when is someone mapping in the field – they could probably see there is a bolted route, but the name is not usually written there. It is maintained by the local climbing organization, usually appears in printed guides, and sometimes there is a signboard nearby with the names, difficulties and a map (topo).

I added the suggested fields to `fields` property.

zbycz commented on data/presets/climbing/route_bottom.json:
```diff
@@ -0,0 +1,33 @@
+        "description"
+    ],
+    "terms": [
+        "rock climbing route"
```
added those + "climbing crag"

zbycz commented:
> In general, it would be great if you could take this opportunity and check if the Wiki pages are beginner friendly already. I thing they need to be improved a bit.
> 
> When I looked at https://wiki.openstreetmap.org/wiki/Tag:sport%3Dclimbing I did not find any mention of the fields here. The fact that there is https://wiki.openstreetmap.org/wiki/Climbing is only visible if users actually read line by line. And only on https://wiki.openstreetmap.org/wiki/Climbing are there more details on what we are doing here.
> 
> Adding some of the fields here as redirects will help. But also adding some more links and callouts to the pages will help. The German page is also bigger than the english page which I always find weird for something that is international https://wiki.openstreetmap.org/wiki/DE:Tag:sport%3Dclimbing (and not scoped to one country in the presets here).

I changed this on the wiki + created correct wikitiems to all of these pages:
- added page for https://wiki.openstreetmap.org/wiki/Tag:climbing%3Droute_bottom - listed all common values and some more from the Climbing page. 
- all keys/tags from the route_bottom page are created:
  - https://wiki.openstreetmap.org/wiki/Key:climbing:length
  - https://wiki.openstreetmap.org/wiki/Key:climbing:bolts
  - https://wiki.openstreetmap.org/wiki/Key:climbing:sport
  - https://wiki.openstreetmap.org/wiki/Key:climbing:trad
  - https://wiki.openstreetmap.org/wiki/Key:climbing:grade:uiaa
  - https://wiki.openstreetmap.org/wiki/Key:climbing:grade:french
  - https://wiki.openstreetmap.org/wiki/Key:climbing:grade:saxon
  - https://wiki.openstreetmap.org/wiki/Key:climbing:grade:yds_class
- i removed the misleading table with `leisure=sport_center` on the https://wiki.openstreetmap.org/wiki/Tag:sport%3Dclimbing page, and added what tags is used for differentiating among the climbing features. The details are left on the large Climbing page. In german the sport=climbing page basically just contains what is in english on the Climbing page. I don't see an issue with that. 
- I added a category https://wiki.openstreetmap.org/wiki/Category:Climbing

tordans commented on data/fields/climbing/grade/french.json:
```diff
@@ -0,0 +1,45 @@
+    "type": "combo",
+    "label": "Grade (French)",
+    "placeholder": "5c",
+    "options": [
```
```suggestion
    "caseSensitive": false,
    "customValues": true,
    "autoSuggestions": false,
    "options": [
```

tordans commented on data/fields/climbing/grade/saxon.json:
```diff
@@ -0,0 +1,36 @@
+    "type": "combo",
+    "label": "Grade (Saxon)",
+    "placeholder": "VIIa",
+    "options": [
```
```suggestion
    "caseSensitive": true,
    "customValues": true,
    "autoSuggestions": false,
    "options": [
```

tordans commented on data/fields/climbing/grade/uiaa.json:
```diff
@@ -0,0 +1,48 @@
+    "type": "combo",
+    "label": "Grade (UIAA)",
+    "placeholder": "6-",
+    "options": [
```
```suggestion
    "caseSensitive": false,
    "customValues": true,
    "autoSuggestions": false,
    "options": [
```

tordans reviewed (changes requested):
The combo field can be improved a bit by adding a few more flags; some are defaults but it makes it more explicit. More at https://github.com/ideditor/schema-builder?tab=readme-ov-file#autosuggestions

I will apply those right away to see them in action.

tordans commented on data/fields/climbing/grade/yds_class.json:
```diff
@@ -0,0 +1,42 @@
+    "type": "combo",
+    "label": "Grade (YDS)",
+    "placeholder": "5.10a",
+    "options": [
```
(Missed one with my last change)

```suggestion
    "caseSensitive": false,
    "customValues": true,
    "autoSuggestions": false,
    "options": [
```

tordans reviewed (approved):
I think this is ready to be merged now.
I changed the combo fields a bit with my last two commits. The Grade (Saxon) requires the `caseSensitive=true` flag. And with explicit `autosuggest=off` the options are sorted just as they are specified in the file and also no network request is done.

To me, this looks good. Shall we merge?
